CMakeLists.txt

```
@@ -72,7 +72,7 @@ set( target-bits-split "standard" CACHE STRING "Split of the 64-bit target neuro
 
 # generic build configuration
 option( static-libraries "Build static executable and libraries [default=OFF]" OFF )
-set( with-optimize ON CACHE STRING "Enable user defined optimizations [default=OFF (uses '-O2')]. When ON, '-O3' is used. Separate multiple flags by ';'." )
+set( with-optimize ON CACHE STRING "Enable user defined optimizations [default=OFF (uses '-O0')]. When ON, '-O3' is used. Separate multiple flags by ';'." )
```

```suggestion
set( with-optimize ON CACHE STRING "Enable user defined optimizations [default=ON (uses '-O2')]. When OFF, no '-O' flag is passed to the compiler. Explicit compiler flags can be given; separate multiple flags by ';'." )
```

The code in `ProcessOptions.cmake` 

https://github.com/nest/nest-simulator/blob/d2500ca3ee16930c0b372694ec90bfe22cb3461f/cmake/ProcessOptions.cmake#L24

indicates that `-Dwith-optimize=ON` will use `-O2`, while `...=OFF` will add no O-option at all. If `-Dwith-optimize` is not given at all, `-O2` is also added. I confirmed these behaviors with a test. What the compiler does when used without any -O flag is up to the compiler and e.g., the Clang man page does not provide information about this.